cordyceps/src/list.rs

```diff
@@ -494,7 +494,7 @@ impl<T: Linked<Links<T>> + ?Sized> List<T> {
         let tail_links = unsafe { T::links(tail) };
         let head_links = unsafe { head_links.as_ref() };
         let tail_links = unsafe { tail_links.as_ref() };
-        if head == tail {
+        if core::ptr::addr_eq(head.as_ptr(), tail.as_ptr()) {
```

Comment:
nit, take it or leave it: `core::ptr` is already imported in this file: https://github.com/hawkw/mycelium/blob/ed9e49b520b574298ae777a06bd8b5fc125c05ac/cordyceps/src/list.rs#L13

so we could drop the `core` prefix:
```suggestion
        if ptr::addr_eq(head.as_ptr(), tail.as_ptr()) {
```